src/pages/spawn/spawnHost/SpawnHostModal.tsx

```diff
@@ -189,9 +203,9 @@ export const SpawnHostModal: React.FC<SpawnHostModalProps> = ({
     dispatch({
       type: "editDistro",
       distroId: d,
-      isVirtualWorkstation: !!virtualWorkstationDistros.find(
-        (vd) => d === vd.name
-      ),
+      isVirtualWorkstation: !!distrosData?.distros.find(
+        (vd) => distroId === vd.name
+      )?.isVirtualWorkStation,
```

Comment:
I'm okay with this actually. The subset you are referring to includes all virtual workstations aside from the current selection. This could seem like a stretch, but I'd like to avoid an edgecases that could result in the same virtual workstation distro being selected twice in a row and on the second time doesn't register as a virtual workstation bc we filtered out the current selection.

Reply:
@khelif96 I realized that the smaller list is filtered to match the current distroInput. that means if the "editDistro" will often set isVirtualWorkstation to false if this function was called twice in a row with differing distroIds. 
However, I realized that the useEffect introduced in this file to recalc isVirtualWorkstation would correct the form state